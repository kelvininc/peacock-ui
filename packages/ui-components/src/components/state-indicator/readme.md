# *<kv-state-indicator>*



<!-- Auto Generated Below -->


## Usage

### Angular / javascript

```html
<!-- Default -->
<kv-state-indicator text="State Indicator" />

<!-- With Color -->
<kv-state-indicator text="State Indicator" color="green" />
```


### React

```tsx
import React from 'react';

import { KvStateIndicator } from '@kelvininc/react-ui-components';

export const StateIndicatorExample: React.FC = () => (
  <>
    {/*-- Default --*/}
	<KvStateIndicator text="State Indicator" />
	
	{/*-- With Color --*/}
	<KvStateIndicator text="State Indicator" color="green" />
  </>
);

```


### Stencil

```tsx
import { Component, h } from '@stencil/core';

@Component({
  tag: 'state-indicator-example',
  styleUrl: 'state-indicator-example.css',
  shadow: true,
})
export class StateIndicatorExample {
  render() {
    return [
      	// Default
		<kv-state-indicator text="State Indicator" />
		
		// With Color
		<kv-state-indicator text="State Indicator" color="green" />
    ];
  }
}
```



## Properties

| Property | Attribute | Description                      | Type     | Default     |
| -------- | --------- | -------------------------------- | -------- | ----------- |
| `color`  | `color`   | (optional) State indicator color | `string` | `undefined` |
| `text`   | `text`    | (optional) State indicator text  | `string` | `undefined` |


----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
