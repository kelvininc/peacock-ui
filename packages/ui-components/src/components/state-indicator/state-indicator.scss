@import '../../assets/styles/globals';

.state-indicator {
	display: flex;
	align-items: baseline;

	.color {
		width: 8px;
		height: 8px;
		border-radius: 50%;
		margin-right: $spacing-2x;
	}

	.text {
		@include kv-font-label-small-regular();
		color: kv-color('text');
	}
}
