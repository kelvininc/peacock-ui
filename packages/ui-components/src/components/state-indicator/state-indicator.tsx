import { Component, Prop, h, Host } from '@stencil/core';

@Component({
	tag: 'kv-state-indicator',
	styleUrl: 'state-indicator.scss',
	shadow: true
})
export class KvStateIndicator {
	/** (optional) State indicator color */
	@Prop({ reflect: true }) color: string;
	/** (optional) State indicator text */
	@Prop({ reflect: true }) text: string;

	render() {
		return (
			<Host>
				<div class="state-indicator">
					{this.color && <div class="color" style={{ background: this.color }}></div>}
					<div class="text">{this.text}</div>
				</div>
			</Host>
		);
	}
}
