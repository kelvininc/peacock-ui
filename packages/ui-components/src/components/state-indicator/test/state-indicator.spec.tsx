import { SpecPage } from '@stencil/core/internal';
import { KvStateIndicator } from '../state-indicator';
import { newSpecPage } from '@stencil/core/testing';

describe('Switch Button (unit tests)', () => {
	let page: SpecPage;

	describe('when uses default props', () => {
		beforeEach(async () => {
			page = await newSpecPage({
				components: [KvStateIndicator],
				html: `<kv-state-inidcator text="State Indicator"></kv-state-indicator>`,
			});
		})

		it('should match the snapshot', () => {
			expect(page.root).toMatchSnapshot();
		});
	})

	describe('when has a color', () => {
		beforeEach(async () => {
			page = await newSpecPage({
				components: [KvStateIndicator],
				html: `<kv-state-inidcator text="State Indicator" color="green"></kv-state-indicator>`,
			});
		})

		it('should match the snapshot', () => {
			expect(page.root).toMatchSnapshot();
		});
	});
});
