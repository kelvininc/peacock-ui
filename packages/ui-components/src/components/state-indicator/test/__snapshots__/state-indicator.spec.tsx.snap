// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch Button (unit tests) when has a color should match the snapshot 1`] = `
<kv-state-inidcator
  color="green"
  text="State Indicator"
/>
`;

exports[`Switch Button (unit tests) when uses default props should match the snapshot 1`] = `
<kv-state-inidcator
  text="State Indicator"
/>
`;
